src/java.desktop/macosx/classes/sun/lwawt/macosx/CAccessibility.java

```diff
@@ -856,6 +858,41 @@ public Accessible call() throws Exception {
         }, c);
     }
 
+    private static Accessible getCurrentAccessiblePopupMenu(Accessible a, Component c) {
+        if (a == null) return null;
+
+        return invokeAndWait(new Callable<Accessible>() {
+            @Override
+            public Accessible call() throws Exception {
+                return traversePopupMenu(a);
+            }
+        }, c);
+    }
+
+    private static Accessible traversePopupMenu(Accessible a) {
+        // a is root level popupmenu
+        AccessibleContext ac = a.getAccessibleContext();
+        if (ac != null) {
+            Accessible aComp = null;
+            for (int i = 0; i < ac.getAccessibleChildrenCount(); i++) {
```

Comment:
I suggest replacing for with while. If the desired submenu is at the beginning of the list, there’s no need to traverse it further.

Reply:
Optimized the code to not traverse further if the desired submenu is fetched. Verified the recent change with the test code.